### ISSUE TITLE
Event/trigger to know if user has scrolled to the bottom of the document.

### ISSUE THREAD

**Issue by qa1453** [stephanrauh/ngx-extended-pdf-viewer#281]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
We're using this package to display a Terms and Conditions document and we want to force the user to scroll through the entire PDF. Looking for an event to trigger if the user has scrolled to the last page in the PDF. 

**Describe alternatives you've considered**
Other alternatives are outside of the pdf viewer package. 

**Additional context**
none

**Comment by stephanrauh** [comment]
Maybe you can achieve your goal with the API that already exists. There's the two-way binding even `[(page)]`.  If you know how many pages your document has, it's easy to check if the user has scrolled down to the last page - to taken their time to scroll through each page individually, if you insist on doing so.

Remains the question how many pages the PDF file has. That's reported by `(pdfLoaded)`.

I guess that's all you need to solve your issue. Does this serve your needs?

**Comment by stephanrauh** [comment]
BTW, I've labeled this issue as "won't fix" because I don't want to do anything. But you can still convince me to do something about the issue. Just tell me!

**Comment by qa1453** [comment]
Yes, I believe this will suit our needs!

Thank you for the succinct and perfect response!

Tom

Tom Hoffman | Software Engineer
thomas.hoffman@infinite.com<mailto:thomas.hoffman@infinite.com> |M: +1 224.355.5077
Communicate with me securely via

[cid:image001.png@01D5A46E.99120330]

NetSfere: https://netsfere.me/thoffman

Infinite Convergence Solutions <http://www.infinite-convergence.com/> | Enabling Communication

**Comment by qa1453** [comment]
Stephan,

I believe the solution you provided will suit our needs, so closing it with a status of “won’t fix” is okay.

Thank you,

Tom

Tom Hoffman | Software Engineer
thomas.hoffman@infinite.com<mailto:thomas.hoffman@infinite.com> |M: +1 224.355.5077
Communicate with me securely via

[cid:image001.png@01D5A46E.99120330]

NetSfere: https://netsfere.me/thoffman

Infinite Convergence Solutions <http://www.infinite-convergence.com/> | Enabling Communication

**Comment by stephanrauh** [comment]
Whow. Sounds like I've managed to help you. That's what makes being an open-source author so much fun!

I'm going to close the issue now. As usual, I'll keep listening to the channel, so don't hesitate to post something here. 

Best regards,
Stephan

**Comment by stephanrauh** [comment]
Hi Tom,

slightly off-topic: being the curious person I am, I followed your corporate links. In general, I like you corporate look and feel. It's a bit weird that your company logo strongly resembles that logo of my employer, but I think that's just a co-incidence. However, I felt several of your web pages are on the slow side. http://www.infinite-convergence.com/ seems to be pretty fast, but https://www.infinite.com tends to be a bit slow.

Maybe that's because all my neighbors are look Netflix. :) But if you feel there's something to improve, don't hesitate to tell my. My employer (including me, in particular, but not only) have some experience in improving web site performance. That, in turn, is important to give you the opportunity to gain a fair Google page rank.

Just a suggestion - if you ignore me, there are no hard feelings!

Best regards,
Stephan

**Comment by qa1453** [comment]
Stephan,

Not going to ignore you!

The faster page you mentioned http://www.infinite-convergence.com/ is more recently redone.  I work for infinite convergence, and we’re a subsidiary of infinite, the slower page.   The parent company is the IT company and should be better at it than us… Though to be fair, I don’t know who did either of those two web pages. I focus on the web pages and entire application for the products we sell, not the sites you’re referring to.

My company was a spin-off of Motorola 10 years ago, and we did very little web development. Most of our products were used by carriers, and the UI wasn’t that important to them. What was important was the reliability and availability of the application that had to be running 24x7. No exceptions.  It wasn’t until the last few years as we try to branch out to other businesses are we starting to focus more and more on the Front End.   I taught myself FE work, including Angular, and am learning as I go. It’s difficult because the FE world changes so fast, and I have to worry about the entire application, not just the backend.

If I controlled the purse strings, I would hire people like you to come in and help us get started correctly. So we don’t make costly mistakes up-front. Unfortunately, our mgmt. doesn’t see things that way, and I don’t control the purse strings…

Thanks for your support on the issue/request I raised. I appreciate the quick response!

Have a good weekend,

Tom

Tom Hoffman | Software Engineer
thomas.hoffman@infinite.com<mailto:thomas.hoffman@infinite.com> |M: +1 224.355.5077
Communicate with me securely via

[cid:image001.png@01D5A46E.99120330]

NetSfere: https://netsfere.me/thoffman

Infinite Convergence Solutions <http://www.infinite-convergence.com/> | Enabling Communication
